lib/syntax_tools/src/epp_dodger.erl

```diff
@@ -796,36 +797,69 @@ rewrite(Node) ->
 		_ ->
 		    Node
 	    end;
-	application ->
-	    F = erl_syntax:application_operator(Node),
-	    case erl_syntax:type(F) of
-		atom ->
-		    case erl_syntax:atom_value(F) of
-			?macro_call ->
-			    [A | As] = erl_syntax:application_arguments(Node),
-			    M = erl_syntax:macro(A, rewrite_list(As)),
-			    erl_syntax:copy_pos(Node, M);
-			_ ->
-			    rewrite_1(Node)
-		    end;
-		_ ->
-		    rewrite_1(Node)
-	    end;
-	_ ->
-	    rewrite_1(Node)
+    application ->
+        rewrite_application(
+            Node,
+            fun erl_syntax:application_operator/1,
+            fun erl_syntax:application_arguments/1);
+    type_application ->
+        rewrite_application(
+            Node,
+            fun erl_syntax:type_application_name/1,
+            fun erl_syntax:type_application_arguments/1);
+    user_type_application ->
+        rewrite_application(
+            Node,
+            fun erl_syntax:user_type_application_name/1,
+            fun erl_syntax:user_type_application_arguments/1);
+    attribute ->
+        Name = erl_syntax:attribute_name(Node),
+        case catch erl_syntax:concrete(Name) of
```

Comment:
I think this can be simplified a bit to both avoid an unnecessary `catch` and make it clear what kind of node are allowed as the attribute name, i.e. an atom node.

```suggestion
        case erl_syntax:atom_value(Name) of
```